torch/signal/windows/windows.py

```diff
@@ -0,0 +1,242 @@
+                     device=device,
+                     requires_grad=requires_grad)
+
+    return torch.exp(-torch.abs(k) / tau)
+
+
+def cosine(window_length: int,
+           periodic: bool = True,
+           dtype: _dtype = None,
+           layout: _layout = torch.strided,
+           device: _device = None,
+           requires_grad: bool = False) -> Tensor:
+    r"""Computes a window with a simple cosine waveform.
+    Also known as the sine window.
+
+    The cosine window is defined as follows:
+
+    .. math::
+        w(n) = \cos{\left(\frac{\pi n}{M} - \frac{\pi}{2}\right)} = \sin{\left(\frac{\pi n}{M}\right)}
+
+    Where `M` is the window length.
+
+    Args:
+        window_length (int): the length of the output window.
+            In other words, the number of points of the cosine window.
+        periodic (bool, optional): If `True`, returns a periodic window suitable for use in spectral analysis.
+            If `False`, returns a symmetric window suitable for use in filter design. Default: `True`.
+
+    Keyword args:
+        dtype (:class:`torch.dtype`, optional): the desired data type of returned tensor.
+            Default: if ``None``, uses a global default (see :func:`torch.set_default_tensor_type`).
+        layout (:class:`torch.layout`, optional): the desired layout of returned window tensor.
+            Only `torch.strided` (dense layout) is supported.
+        device (:class:`torch.device`, optional): the desired device of returned tensor.
+            Default: if ``None``, uses the current device for the default tensor type
+            (see :func:`torch.set_default_tensor_type`).
+            :attr:`device` will be the CPU for CPU tensor types and the current CUDA device for CUDA tensor types.
+        requires_grad (bool, optional): If autograd should record operations on the returned tensor. Default: ``False``.
+
+    Examples:
+        >>> # Generate a cosine window without keyword args.
+        >>> torch.signal.windows.cosine(10)
+        tensor([0.1423, 0.4154, 0.6549, 0.8413, 0.9595, 1.0000, 0.9595, 0.8413, 0.6549,
+        0.4154])
+
+        >>> # Generate a symmetric cosine window.
+        >>> torch.signal.windows.cosine(10,periodic=False)
+        tensor([0.1564, 0.4540, 0.7071, 0.8910, 0.9877, 0.9877, 0.8910, 0.7071, 0.4540,
+        0.1564])
+    """
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('cosine_window', window_length, dtype, layout)
+
+    if window_length == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if window_length == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if periodic:
+        window_length += 1
+
+    k = torch.arange(window_length, dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+    window = torch.sin(torch.pi / window_length * (k + .5))
```

Comment:
Note that here you are transforming `k` using an affine transformation. The `start, stop, step` parameters allow you to model affine transformations a sequence. More mathematically, affine transformations take arithmetic sequences to arithmetic sequences

Reply:
Updated the cosine as well now.

Reply:
We can still do better. Note that you do:
```
return torch.sin(torch.pi / (window_length + 1 if periodic else window_length) * k)
```
So you are multiplying `k` by a constant. That you can fit into the `step` parameter, as it's an affine transformation. This function should just call `arange` and do a `return torch.sin(k)`